### PR TITLE
feat(mini-rx-store): selector with dictionary parameter

### PR DESCRIPTION
**Pull request by spierala**
Shamefully copied from NgRx. But we need this! :)

See the orignal NgRx PR here: https://github.com/ngrx/platform/pull/3703

### CODE REVIEW

File: apps/mini-rx-angular-demo/src/app/modules/todos/components/todos-shell/todos-shell.component.html [hunk 1]
```diff
@@ -41,13 +42,13 @@
...
-                        [todos]="(todosDone$ | async)!"
-                        [selectedTodo]="(selectedTodo$ | async)!"
+                        [todos]="viewModel.todosDone"
+                        [selectedTodo]="viewModel.selectedTodo"
                         (selectTodo)="todosState.selectTodo($event)"
                     ></app-todo-list>
                 </div>
             </div>
-            <div class="col" *ngIf="selectedTodo$ | async as selectedTodo">
+            <div class="col" *ngIf="viewModel.selectedTodo as selectedTodo">
```
Comment (vangenechtenbert) [inline apps/mini-rx-angular-demo/src/app/modules/todos/components/todos-shell/todos-shell.component.html:51]: Do you still need to cast objec type?

Comment (spierala) [reply apps/mini-rx-angular-demo/src/app/modules/todos/components/todos-shell/todos-shell.component.html:51]: Do you mean the "ngIf as" syntax?

It works also without Observables.

The alternative would be:

image

Which one do you prefer?

Comment (vangenechtenbert) [reply apps/mini-rx-angular-demo/src/app/modules/todos/components/todos-shell/todos-shell.component.html]: Both should be fine

**Review by vangenechtenbert — APPROVED** [review]

File: apps/mini-rx-angular-demo/src/app/modules/todos/components/todos-shell/todos-shell.component.html (hunk 1 above)
Comment (spierala) [reply apps/mini-rx-angular-demo/src/app/modules/todos/components/todos-shell/todos-shell.component.html]: Removed the ngIf as syntax.

It has indeed no benefit here.

**Comment by spierala** [comment]
@all-contributors please add @vangenechtenbert for review.

**Pull request merged**